src/main/java/com/mixpanel/android/mpmetrics/PersistentIdentity.java

```diff
@@ -579,7 +618,7 @@ protected void removeOptOutFlag(String token) {
         try {
             final SharedPreferences prefs = mMixpanelPreferences.get();
             final SharedPreferences.Editor prefsEditor = prefs.edit();
-            prefsEditor.clear();
+            prefsEditor.remove("opt_out_" + token);
```

Comment:
Fix this bug that cleared all preferences when removing the opt out flag for a given project token.